src/components/learner-credit-management/data/utils.js

```diff
@@ -680,36 +682,50 @@ export const getAssignableCourseRuns = ({ courseRuns, subsidyExpirationDatetime,
     enrollStart: courseRun.hasEnrollStart ? dayjs.unix(courseRun.enrollStart).toISOString() : null,
     upgradeDeadline: dayjs.unix(courseRun.upgradeDeadline).toISOString(),
   }));
+
   const assignableCourseRunsFilter = ({
-    enrollBy, enrollStart, start, isActive, hasEnrollBy, isLateEnrollmentEligible,
+    enrollBy, enrollStart, start, hasEnrollBy, hasEnrollStart, isActive, isLateEnrollmentEligible,
   }) => {
+    const isWithinEnrollByDateThreshold = isEnrollByDateWithinThreshold({
+      hasEnrollBy,
+      enrollBy,
+      isLateRedemptionAllowed,
+    });
+    const isWithinStartDateThreshold = isStartDateWithinThreshold({
+      hasEnrollStart,
+      enrollStart,
+      start,
+      subsidyExpirationDatetime,
+    });
+
     // Determine eligibility based on the provided enrollBy, start, and enrollStart date
-    const isEligibleForEnrollment = isEnrollByDateWithinThreshold(
-      {
-        enrollBy,
-        isLateRedemptionAllowed,
-      },
-    ) && isStartDateWithinThreshold(
-      {
-        enrollStart,
-        start,
-        subsidyExpirationDatetime,
-      },
-    );
-    if (hasEnrollBy && isLateRedemptionAllowed && isDateBeforeToday(enrollBy)) {
-      return isLateEnrollmentEligible && isEligibleForEnrollment;
+    const isEligibleForEnrollment = isWithinEnrollByDateThreshold && isWithinStartDateThreshold;
+
+    if (!isEligibleForEnrollment) {
+      // Basic checks against this content's critical dates and their relation to
+      // the current date and subsidy expiration date have failed.
+      return false;
+    }
+    if (isLateRedemptionAllowed) {
```

Comment:
[clarification] Is this intentionally no longer relying on `hasEnrollBy && isDateBeforeToday(enrollBy)`?

Reply:
Updated to reinclude `if (hasEnrollBy && isLateRedemptionAllowed && isDateBeforeToday(enrollBy))`
I had missed that if `hasEnrollBy` is false, we return true regardless, so this check is required to return late enrollment runs 👍🏽